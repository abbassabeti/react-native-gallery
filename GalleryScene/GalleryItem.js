// Imports: Dependencies
import React from 'react';
import {Image } from 'react-native';
import {Dimensions, TouchableOpacity, StyleSheet, Text } from 'react-native';
//import CacheableImage from 'react-native-cacheable-image'

// Screen Dimensions
const { height, width } = Dimensions.get('window');
// Screen: Infinite Scroll

export default GalleryItem = (id,title,navigateTo) => {
  let imgId = id && id.id ? id.id : id
  let imgTitle = id && id.title ? id.title : title
  let navFunc = id && id.navigateTo ? id.navigateTo : navigateTo
  return (
      <TouchableOpacity
        testID={`touchable${imgId}`}
        style={styles.item}
        onPress={()=> navFunc("image",{imageId:imgId,imgTitle})}>
        {/*<CacheableImage
            style={styles.image}
            source={{uri:`https://picsum.photos/id/${id}/200`}}
            checkNetwork
            networkAvailable
        >
        </CacheableImage>*/}
          <Image
            style={styles.image}
            id={imgId}
          source={{uri:`https://picsum.photos/id/${imgId}/200`}}>
          </Image>
      </TouchableOpacity>
    )};

  // Styles
const styles = StyleSheet.create({
  item: {
    backgroundColor: '#cccccc',
    marginVertical: 8,
    marginHorizontal: 16,
    flex: 1,
    aspectRatio: 1,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: 'transparent',
    overflow: 'hidden'
  },
  image:{
    width: '100%',
    height: '100%',
  },
});