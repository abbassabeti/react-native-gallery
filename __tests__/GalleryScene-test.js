import React from 'react';
import ReactDOM from 'react-dom';
import { fireEvent, render, waitFor } from '@testing-library/react-native'
import GalleryScene from '../GalleryScene/GalleryScene'
import GalleryItem from '../GalleryScene/GalleryItem';

//TODO: needs some more improvments for testing scrolling behavior
describe('Testing GalleryScene', () => {
    it(`Sth is wrong with Gallery Snapshot or scrolling stuff`, async() => {
      const onLoadMore = jest.fn();
      const galleryRef = React.createRef()
      const componentTree = render(<GalleryScene ref={galleryRef} notifyLoadMore={onLoadMore}/>)
      expect(componentTree).toMatchSnapshot()
      const {getByTestId} = componentTree
      const list = getByTestId('scrollable-list')
      expect(list).not.toBeNull()
      let element = galleryRef.current
      let initialCount = element.state.documentData.length

      element.scrollToBottom()
      let i = 9
      await waitFor(()=>{
        if (i < 36){
          i = i + 3
        }
        let elem = getByTestId(`touchable${i}`)
        expect(elem)
      },{interval:100,timeout: 4500})
    })
})
