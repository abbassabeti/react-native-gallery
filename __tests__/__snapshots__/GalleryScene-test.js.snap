// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FlatList Sth is wrong with Gallery Snapshot or scrolling stuff 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "height": 1334,
      "paddingBottom": 125,
      "width": 750,
    }
  }
>
  <RCTScrollView
    ListFooterComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={
      Array [
        Object {
          "id": 1,
          "title": "sth",
        },
        Object {
          "id": 2,
          "title": "sth",
        },
        Object {
          "id": 3,
          "title": "sth",
        },
        Object {
          "id": 4,
          "title": "sth",
        },
        Object {
          "id": 5,
          "title": "sth",
        },
        Object {
          "id": 6,
          "title": "sth",
        },
        Object {
          "id": 7,
          "title": "sth",
        },
        Object {
          "id": 8,
          "title": "sth",
        },
        Object {
          "id": 9,
          "title": "sth",
        },
        Object {
          "id": 10,
          "title": "sth",
        },
        Object {
          "id": 11,
          "title": "sth",
        },
        Object {
          "id": 12,
          "title": "sth",
        },
        Object {
          "id": 13,
          "title": "sth",
        },
        Object {
          "id": 14,
          "title": "sth",
        },
        Object {
          "id": 15,
          "title": "sth",
        },
        Object {
          "id": 16,
          "title": "sth",
        },
        Object {
          "id": 17,
          "title": "sth",
        },
        Object {
          "id": 18,
          "title": "sth",
        },
        Object {
          "id": 19,
          "title": "sth",
        },
        Object {
          "id": 20,
          "title": "sth",
        },
        Object {
          "id": 21,
          "title": "sth",
        },
        Object {
          "id": 22,
          "title": "sth",
        },
        Object {
          "id": 23,
          "title": "sth",
        },
        Object {
          "id": 24,
          "title": "sth",
        },
        Object {
          "id": 25,
          "title": "sth",
        },
        Object {
          "id": 26,
          "title": "sth",
        },
        Object {
          "id": 27,
          "title": "sth",
        },
        Object {
          "id": 28,
          "title": "sth",
        },
        Object {
          "id": 29,
          "title": "sth",
        },
        Object {
          "id": 30,
          "title": "sth",
        },
        Object {
          "id": 31,
          "title": "sth",
        },
        Object {
          "id": 32,
          "title": "sth",
        },
        Object {
          "id": 33,
          "title": "sth",
        },
        Object {
          "id": 34,
          "title": "sth",
        },
        Object {
          "id": 35,
          "title": "sth",
        },
        Object {
          "id": 36,
          "title": "sth",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="scrollable-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 36,
              "fontWeight": "600",
              "marginBottom": 12,
              "marginLeft": 12,
            }
          }
        >
          Images
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable1"
          >
            <Image
              id={1}
              source={
                Object {
                  "uri": "https://picsum.photos/id/1/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable2"
          >
            <Image
              id={2}
              source={
                Object {
                  "uri": "https://picsum.photos/id/2/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable3"
          >
            <Image
              id={3}
              source={
                Object {
                  "uri": "https://picsum.photos/id/3/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable4"
          >
            <Image
              id={4}
              source={
                Object {
                  "uri": "https://picsum.photos/id/4/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable5"
          >
            <Image
              id={5}
              source={
                Object {
                  "uri": "https://picsum.photos/id/5/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable6"
          >
            <Image
              id={6}
              source={
                Object {
                  "uri": "https://picsum.photos/id/6/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable7"
          >
            <Image
              id={7}
              source={
                Object {
                  "uri": "https://picsum.photos/id/7/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable8"
          >
            <Image
              id={8}
              source={
                Object {
                  "uri": "https://picsum.photos/id/8/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable9"
          >
            <Image
              id={9}
              source={
                Object {
                  "uri": "https://picsum.photos/id/9/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable10"
          >
            <Image
              id={10}
              source={
                Object {
                  "uri": "https://picsum.photos/id/10/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable11"
          >
            <Image
              id={11}
              source={
                Object {
                  "uri": "https://picsum.photos/id/11/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable12"
          >
            <Image
              id={12}
              source={
                Object {
                  "uri": "https://picsum.photos/id/12/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable13"
          >
            <Image
              id={13}
              source={
                Object {
                  "uri": "https://picsum.photos/id/13/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable14"
          >
            <Image
              id={14}
              source={
                Object {
                  "uri": "https://picsum.photos/id/14/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable15"
          >
            <Image
              id={15}
              source={
                Object {
                  "uri": "https://picsum.photos/id/15/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable16"
          >
            <Image
              id={16}
              source={
                Object {
                  "uri": "https://picsum.photos/id/16/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable17"
          >
            <Image
              id={17}
              source={
                Object {
                  "uri": "https://picsum.photos/id/17/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable18"
          >
            <Image
              id={18}
              source={
                Object {
                  "uri": "https://picsum.photos/id/18/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable19"
          >
            <Image
              id={19}
              source={
                Object {
                  "uri": "https://picsum.photos/id/19/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable20"
          >
            <Image
              id={20}
              source={
                Object {
                  "uri": "https://picsum.photos/id/20/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable21"
          >
            <Image
              id={21}
              source={
                Object {
                  "uri": "https://picsum.photos/id/21/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable22"
          >
            <Image
              id={22}
              source={
                Object {
                  "uri": "https://picsum.photos/id/22/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable23"
          >
            <Image
              id={23}
              source={
                Object {
                  "uri": "https://picsum.photos/id/23/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable24"
          >
            <Image
              id={24}
              source={
                Object {
                  "uri": "https://picsum.photos/id/24/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable25"
          >
            <Image
              id={25}
              source={
                Object {
                  "uri": "https://picsum.photos/id/25/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable26"
          >
            <Image
              id={26}
              source={
                Object {
                  "uri": "https://picsum.photos/id/26/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable27"
          >
            <Image
              id={27}
              source={
                Object {
                  "uri": "https://picsum.photos/id/27/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable28"
          >
            <Image
              id={28}
              source={
                Object {
                  "uri": "https://picsum.photos/id/28/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable29"
          >
            <Image
              id={29}
              source={
                Object {
                  "uri": "https://picsum.photos/id/29/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#cccccc",
                "flex": 1,
                "marginHorizontal": 16,
                "marginVertical": 8,
                "opacity": 1,
              }
            }
            testID="touchable30"
          >
            <Image
              id={30}
              source={
                Object {
                  "uri": "https://picsum.photos/id/30/200",
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
      <View
        onLayout={[Function]}
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
